### DISCUSSION TITLE
FRSKY ADV Sensor Support + New Power Switch

### DISCUSSION THREAD

**Discussion by m1n91** [EdgeTX/edgetx#3665] [General]
Hello Chaps, and many thanks for your hard work so far.

Support for FRSKY's new ADV sensors and POWER SWITCH.
The data does come through in Telemetry, but in a completely wrong formats.

I approached FRSKY about this and they said to wait for a new firmware version to support this.

The ADV PS30 (air pressure sensor) comes through as 0E60 and pressure in millibars but to six digits 101345. The correct pressure would be 1013. 

To fix this; The RATIO field within the telemetry submenu also appears to be broken.

On edge tx companions simulator it works correctly but not on the transmitter 😕

The power switch comes through with a load of garbled numbers which make no sense what so ever.

**Comment by aspence** [comment]
hey @m1n91 !

did you get any ADV sensors to work? I'm working on same - trying to get a FAS40ADV plugged into an Archer Plus SR8 to give current and voltage telemetry over serial (FBUS or S.Port or whatever protocol) and out to my Radiomaster TX16S running EdgeTx 2.10.3.

i've got the AP LUA scripts from FrSky that work for stabilitzation setup. haven't tried the calibration one, but it might work. but the setup one didn't have the "Telemetry" setting for the reciever, so i couldnt try changing the telemetry protocol on the SPort.

THe sensor flashes what I assume is "Fast" when i plug it in - which the manual says in FBUS protocol. Maybe my reciever defaults to S.Bus.

Either way with it runinng and sensor flashing if i do "Discover new" on the telemetry on the radio no new telemetry settings come up.

So your issue above would seem very relevant!

Thanks for any help!

**Reply by aspence** [reply]
Sorted!! I'm a dummy. I had the sensor plugged into the S.Port slot for redundant recievers/training. It needed to be in the 4-pin breakout for FBUS/AIN. Used the breakout and VOILA! Getting lovely current and voltage telemetry. Cheers!
